querybook/server/lib/query_analysis/templating.py

```diff
@@ -26,179 +30,241 @@ class QueryJinjaSyntaxException(QueryTemplatingError):
     pass
 
 
+class LatestPartitionFunctionException(QueryTemplatingError):
+    pass
+
+
 # The first part is regex for single line comment, ie -- some comment
 # the second part is for multi line comment, ie /* test */
 comment_re = re.compile(r"((?:--.*)|(?:\/\*(?:.|\n)*?\*\/))", re.MULTILINE)
 
 
-def _escape_sql_comments(query: str):
-    return re.sub(
-        comment_re, lambda match: "{{ " + json.dumps(match.group()) + " }}", query
-    )
+class TemplatedQueryRenderer(object):
```

Comment:
what would the reasoning to use class here? there is no inheritance and the only function that gets called outside of the module would be render_templated_query? With class the flow is harder to follow with all the object properties

Reply:
I used a class here because the env (or the engine_id) needs to be shared with 4 of the helper functions to be able to use the `latest_partition` global function (so it needs to be added/passed to a bunch of function definitions and function calls). I thought it'd be cleaner to just save it as a class variable - but if that doesn't make sense I can switch it back.